### PR TITLE
feat: add genesis state for datasale

### PR DESCRIPTION
**Pull request by inchori**
- Implement a genesis state for datasale.
- Add `InitGenesis` and `ExportGenesis` function in `x/datadeal/genesis.go`.
- Add `DataSales` in `DefaultGenesis` and `Validate` in `x/datadeal/types/genesis.go`.
- Add `GetAllDataSalesList` in `keeper/deal.go`.
- The unit test code for `InitGenesis` and `ExportGenesis` will be implemented after `CreateDeal` transaction merged.

### CODE REVIEW

File: x/datadeal/keeper/deal.go [hunk 1]
```diff
@@ -77,3 +77,25 @@ func (k Keeper) SetDataSale(ctx sdk.Context, dataSale *types.DataSale) error {
 
 	return nil
 }
+
+func (k Keeper) GetAllDataSaleList(ctx sdk.Context) ([]types.DataSale, error) {
```
Comment (gyuguen) [inline x/datadeal/keeper/deal.go:81]: I wish there was a test for this `get` function.

Comment (inchori) [reply x/datadeal/keeper/deal.go:81]: Add test code in [ccf0429](https://github.com/medibloc/panacea-core/pull/426/commits/ccf0429ae930c9882a51c86d89cfacc010be424c).

**Review by H4NLee — APPROVED** [review]
lgtm

**Review by gyuguen — APPROVED** [review]
LGTM 👍

**Review by audtlr24 — APPROVED** [review]
LGTM!

**Pull request merged**